receiver/vmmetricsreceiver/vm_metrics_collector.go

```diff
@@ -113,104 +103,107 @@ func (vmc *VMMetricsCollector) StopCollection() {
-				mUserCPUSeconds.M(cpuStat.User),
-				mNiceCPUSeconds.M(cpuStat.Nice),
-				mSystemCPUSeconds.M(cpuStat.System),
-				mIdleCPUSeconds.M(cpuStat.Idle),
-				mIowaitCPUSeconds.M(cpuStat.Iowait))
-		}
-		prevStat = &stat
+		metrics = append(
+			metrics,
+			&metricspb.Metric{
+				MetricDescriptor: metricProcessesRunning,
+				Timeseries:       []*metricspb.TimeSeries{vmc.getInt64TimeSeries(stat.ProcessesRunning)},
+			},
+			&metricspb.Metric{
+				MetricDescriptor: metricProcessesBlocked,
+				Timeseries:       []*metricspb.TimeSeries{vmc.getInt64TimeSeries(stat.ProcessesBlocked)},
+			},
+			&metricspb.Metric{
+				MetricDescriptor: metricProcessesCreated,
+				Timeseries:       []*metricspb.TimeSeries{vmc.getInt64TimeSeries(stat.ProcessCreated)},
+			},
+			&metricspb.Metric{
+				MetricDescriptor: metricCPUSeconds,
+				Timeseries: []*metricspb.TimeSeries{
+					vmc.getDoubleTimeSeries(cpuStat.User, labelValueCPUUser),
+					vmc.getDoubleTimeSeries(cpuStat.System, labelValueCPUSystem),
+					vmc.getDoubleTimeSeries(cpuStat.Idle, labelValueCPUIdle),
+					vmc.getDoubleTimeSeries(cpuStat.Nice, labelValueCPUNice),
+					vmc.getDoubleTimeSeries(cpuStat.Iowait, labelValueCPUIOWait),
+				},
+			},
+		)
+	} else {
+		errs = append(errs, err)
 	}
-	return prevProcStat, prevStat
-}
 
-func (vmc *VMMetricsCollector) export() {
-	vds := make([]*view.Data, 0, len(vmc.views))
-	for _, v := range vmc.views {
-		if rows, err := view.RetrieveData(v.Name); err == nil {
-			vd := view.Data{
-				View:  v,
-				Start: vmc.startTime,
-				End:   time.Now(),
-				Rows:  rows,
-			}
-			vds = append(vds, &vd)
-		}
+	if len(errs) > 0 {
+		span.SetStatus(trace.Status{Code: 15 /*DATA_LOSS*/, Message: fmt.Sprintf("Error(s) when scraping VM metrics: %v", errs)})
 	}
-	vmc.uploadViewData(vds)
-}
 
-func (vmc *VMMetricsCollector) uploadViewData(vds []*view.Data) {
-	if len(vds) == 0 {
-		return
+	if len(metrics) > 0 {
+		vmc.consumer.ConsumeMetricsData(ctx, data.MetricsData{Metrics: metrics})
 	}
+}
 
-	ctx, span := trace.StartSpan(context.Background(), "VMMetricsCollector.uploadViewData")
-	defer span.End()
-
-	metrics := make([]*metricspb.Metric, 0, len(vds))
-	errs := []error{}
-	for _, vd := range vds {
-		if metric, err := viewDataToMetric(vd); err == nil {
-			metrics = append(metrics, metric)
-		} else {
-			errs = append(errs, err)
-		}
+func (vmc *VMMetricsCollector) getInt64TimeSeries(val uint64) *metricspb.TimeSeries {
+	return &metricspb.TimeSeries{
+		StartTimestamp: internal.TimeToTimestamp(vmc.startTime),
+		Points:         []*metricspb.Point{{Timestamp: internal.TimeToTimestamp(time.Now()), Value: &metricspb.Point_Int64Value{Int64Value: int64(val)}}},
 	}
-	if len(metrics) > 0 {
-		vmc.consumer.ConsumeMetricsData(ctx, data.MetricsData{Metrics: metrics})
+}
+
+func (vmc *VMMetricsCollector) getDoubleTimeSeries(val float64, labelVal *metricspb.LabelValue) *metricspb.TimeSeries {
+	labelVals := []*metricspb.LabelValue{}
```

Comment:
nit: this slice escapes, but many times is used the empty one, could we have it as nil instead? Or does the `metricspb.TimeSeries` requires it to be non-nil?

Reply:
You're right those don't need to be non-nil - updated.